library/blivet.py

```diff
@@ -1326,19 +1338,21 @@ def _manage_encryption(self, members):
 
     def _manage_thin_pools(self, pool_device):
 
-        def get_vol_size(volume):
-            if isinstance(volume, str) and '%' in volume:
+        def str_to_size(spec, hundredpercent=None):
+            # Convert given 'spec' string to Size. When input string can be in percent
+            # (e.g. '50%'), use the optional parameter (type Size) as a reference
+            if isinstance(spec, str) and ('%' in spec):
                 try:
-                    percentage = int(volume[:-1].strip())
+                    percentage = int(spec[:-1].strip())
                 except ValueError:
-                    raise BlivetAnsibleError("invalid percentage '%s' size specified for volume in pool '%s'" % (volume, pool_device.name))
+                    raise BlivetAnsibleError("invalid percentage '%s' size specified in pool '%s'" % (spec, pool_device.name))
 
-                size = pool_device.size * (percentage / 100.0)
+                size = Size(hundredpercent * (percentage / 100.0))
             else:
                 try:
-                    size = Size(volume)
+                    size = Size(spec)
                 except Exception:
-                    raise BlivetAnsibleError("invalid size specification for volume in pool '%s'" % pool_device.name)
+                    raise BlivetAnsibleError("invalid size specification '%s' in pool '%s'" % (spec, pool_device.name))
```

Comment:
This affects tests_resize.yml because it searches for this regex:
```
        - name: Verify the output
          assert:
            that: "blivet_output.failed and
                   blivet_output.msg|regex_search('invalid size.+for volume') and
                   not blivet_output.changed"
            msg: "Unexpected behavior w/ invalid volume size"
```